src/ape/types/__init__.py

```diff
@@ -415,6 +418,64 @@ def __contains__(self, val: Any) -> bool:
         return any(log == val for log in self)
 
 
+_T = TypeVar("_T")  # _LazySequence generic.
+
+
+class _LazySequence(Sequence[_T]):
```

Comment:
at this time, this is only really used in internal stuff in the framework for performance reasons; i tried using `collections.deque()` but it wasnt lazily enough and so i went with this but if there is way to make deque not iterate at all, that would be nice, or an alternative data structure